routers/api/v1/repo/issue_label.go

```diff
@@ -339,3 +307,28 @@ func ClearIssueLabels(ctx *context.APIContext) {
 
 	ctx.Status(http.StatusNoContent)
 }
+
+func prepareForReplaceOrAdd(ctx *context.APIContext, form api.IssueLabelsOption) (issue *models.Issue, labels []*models.Label, err error) {
+	issue, err = models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	labels, err = models.GetLabelsInRepoByIDs(ctx.Repo.Repository.ID, form.Labels)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "GetLabelsInRepoByIDs", err)
+		return
+	}
+
+	if !ctx.Repo.CanWriteIssuesOrPulls(issue.IsPull) {
```

Comment:
We should check as follows:
- User can't READ this issue?: return 404
- User can't WRITE this issue?: return 403
- Other errors: return accordingly

The READ check should be performed first thing in the function.

Reply:
@guillep2k already here!
https://github.com/go-gitea/gitea/blob/4e912d2ed869809066c98cf4ba3675de8086c29d/routers/api/v1/api.go#L709
is definde in scope of 
https://github.com/go-gitea/gitea/blob/4e912d2ed869809066c98cf4ba3675de8086c29d/routers/api/v1/api.go#L697
witch enfocrce this check:
https://github.com/go-gitea/gitea/blob/4e912d2ed869809066c98cf4ba3675de8086c29d/routers/api/v1/api.go#L736
witch return 404 if permission is not granted
https://github.com/go-gitea/gitea/blob/4e912d2ed869809066c98cf4ba3675de8086c29d/routers/api/v1/api.go#L478

Reply:
beside this I think ths changes are outside of scope of this BUGFIX @guillep2k